### ISSUE TITLE
Ionic Angular Components do not render properly during component testing.

### ISSUE THREAD

**Issue by elio-maggini** [cypress-io/cypress#24226]
### Current behavior

1) Create a new Ionic Angular basic project.
2) Add Cypress 10.9.0 as a dev dependency.
3) Create a basic component test for the auto generated home page.
4) Run the test
5) viewWillEnter lifecycle method does not trigger and ion-content section does not render.

<img width="423" alt="Screen Shot 2022-10-11 at 3 38 26 PM" src="https://user-images.githubusercontent.com/54586689/195369936-04d6361f-6b79-4085-a6cc-b0a62c8266ef.png">
<img width="1281" alt="Screen Shot 2022-10-11 at 3 38 35 PM" src="https://user-images.githubusercontent.com/54586689/195369942-df81b79e-dabf-4f7e-ac25-545cb46c6614.png">

### Desired behavior

Component should render as expected.

### Test code to reproduce

[sample.zip](https://github.com/cypress-io/cypress/files/9765868/sample.zip)

### Cypress Version

10.9.0

### Node version

v18.4.0

### Operating System

macOS 12.6

**Comment by warrensplayer** [comment]
@elio-maggini Thanks so much for the sample code to show the issue. That was super helpful.

While Cypress does not have official support for Ionic apps, you should be able to run component tests just fine. What you do have to keep in mind is that in the component test, the component is getting mounted without being inside your application.  If it relies on other parts of your application to work, then you while have to supply them or mock them.  

For the issue with the `<ion-content>` tag, it is getting a height of `0`.  I came across [this](https://github.com/ionic-team/ionic-framework/issues/14432) old issue which led me to try wrapping the `<HomePage>` component with `<ion-app>` like this:
```
cy.mount('<ion-app><app-home></app-home></ion-app>', {
  declarations: [HomePage],
  imports: [IonicModule.forRoot()]
});
```

That did show the information in the `<ion-content>` section as expected.

As for the `ionViewWillEnter` lifecycle method, that looks to be tied to the routing built into Ionic.  There is a React example in the Cypress docs here that shows how to provide  ReactRouter required dependencies with a custom `mount` command.  That might be a way for you to go.

Another option is to use E2E tests for things that require some of these dependencies instead.

**Comment by elio-maggini** [comment]
@warrensplayer Thank you for the information. I'll update in next 24 hours with further info. Maybe even a small PR to update the docs :)

**Comment by warrensplayer** [comment]
@elio-maggini I hope you have been able to make some progress.  I am closing this issue, but please open a new issue if needed.

**Comment by elio-maggini** [comment]
Thanks. We made some progress but not sufficient. 

There are enough challenges with the approach that our end result is to
continue to use E2e testing via Cypress Studio with fixtures to generate
"feature tests".

---

On 2022-10-19 16:14, Stokes Player wrote:

> Closed #24226 [1] as completed. 
> 
> --
> Reply to this email directly, view it on GitHub [2], or unsubscribe [3].
> You are receiving this because you were mentioned.Message ID: ***@***.***>
 

Links:
------
[1] https://github.com/cypress-io/cypress/issues/24226
[2] https://github.com/cypress-io/cypress/issues/24226#event-7625004647
[3]
